hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/lifecycle/run/RunManager.java

```diff
@@ -111,25 +114,27 @@ public RunResult run() throws DetectUserFriendlyException, InterruptedException,
         RunOptions runOptions = detectConfigurationFactory.createRunOptions();
 
         DetectToolFilter detectToolFilter = runOptions.getDetectToolFilter();
-
+        DetectorEnvironment detectorEnvironment = new DetectorEnvironment(directoryManager.getSourceDirectory(), Collections.emptySet(), 0, null, false);
         logger.info(ReportConstants.RUN_SEPARATOR);
         if (detectToolFilter.shouldInclude(DetectTool.DOCKER)) {
             logger.info("Will include the docker tool.");
-            DockerTool dockerTool = new DockerTool(detectContext);
-
-            DockerToolResult dockerToolResult = dockerTool.run();
-            runResult.addToolNameVersionIfPresent(DetectTool.DOCKER, dockerToolResult.dockerProjectNameVersion);
-            runResult.addDetectCodeLocations(dockerToolResult.dockerCodeLocations);
-            runResult.addDockerFile(dockerToolResult.dockerTar);
-
-            if (dockerToolResult.resultType == DockerToolResult.DockerToolResultType.FAILURE) {
-                eventSystem.publishEvent(Event.ExitCode, new ExitCodeRequest(ExitCodeType.FAILURE_GENERAL_ERROR, dockerToolResult.errorMessage));
-            }
+            ToolRunner toolRunner = new ToolRunner(eventSystem, detectContext.getBean(DockerDetector.class, detectorEnvironment));
```

Comment:
Minor, but you should retrieve the DetectorFactory and ask it to create the detectors. But since these are technically 'not detectors' I don't know what we want.

DetectorFactory detectorFactory = detectContext.getBean(DetectorFactory.class);
detectorFactory.createDocker();